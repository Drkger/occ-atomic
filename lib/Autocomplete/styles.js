import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
