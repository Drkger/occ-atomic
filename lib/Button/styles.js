import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};