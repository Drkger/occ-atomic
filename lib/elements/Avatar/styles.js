export default {
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
};