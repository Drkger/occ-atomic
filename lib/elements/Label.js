import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

const styles = {
    label: {
        boxSizing:'border-box',
        fontFamily:fonts.body,
        fontSize:'11px',
        fontWeight:'normal',
        display:'inline-block',
        height:'22px',
        lineHeight:'20px',
        textAlign:'center',
        borderRadius:'11px',
        padding:'0 15px',
        border:'1px solid',
        position:'relative',
        overflow:'hidden',
        marginRight:'5px'
    },
    cursor: {
        cursor:'pointer'
    },
    default: {
        color:colors.grey8,
        background:colors.white,
        borderColor:colors.grey8
    },
    close: {
        width:'18px',
        height:'18px',
        display:'block',
        position:'absolute',
        top:'1px',
        right:'4px',
        transition:'0.3s all',
        borderRadius:'50%',
        opacity:'0',
        cursor:'pointer',
        '&:hover': {
            opacity:'1'
        },
        '&:before': {
            content:'""',
            width:'50px',
            height:'50px',
            background:colors.white,
            filter:'blur(10px)',
            position:'absolute',
            borderRadius:'50%',
            top:'-16px',
            left:'-16px'
        },
        '&:after': {
            content:'""',
            width:'18px',
            height:'18px',
            background:icons.base(icons.close.icon(['#abb6bd'])),
            position:'absolute',
            top:'0',
            left:'0'
        },
    }
};

class Label extends React.Component {
    
    render() {
        const { classes, children, onClose, onClick, theme } = this.props;
        return (
            <label className={`${classes.label} ${theme ? classes[theme] : classes.default}${onClick ? ` ${classes.cursor}` : ''}`} onClick={onClick}>
                {children}
                {onClose && <span className={classes.close} onClick={e => {e.stopPropagation(); onClose();}} />}
            </label>
        );
    }
}

Label.defaultProps= {
    theme: "default"
};

Label.propTypes = {
    classes: PropTypes.object,
    children: PropTypes.node,
    onClose: PropTypes.func,
    onClick: PropTypes.func
};

export default injectSheet(styles)(Label);