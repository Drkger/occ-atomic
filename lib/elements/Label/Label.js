import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

class Label extends React.Component {
    
    render() {
        const { classes, children, className, id, style, onClose, onClick, theme } = this.props;
        return (
            <label className={`${classes.label} ${theme ? classes[theme] : classes.default}${onClick ? ` ${classes.cursor}` : ''}${className ? ` ${className}` : ''}`} id={id} style={style} onClick={onClick}>
                {children}
                {onClose && <span className={classes.close} onClick={e => {e.stopPropagation(); onClose();}} />}
            </label>
        );
    }
}

Label.defaultProps= {
    theme: "default"
};

Label.propTypes = {
    classes: PropTypes.object,
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string,
    style: PropTypes.object,
    onClose: PropTypes.func,
    onClick: PropTypes.func,
    theme: PropTypes.string
};

export default Label;