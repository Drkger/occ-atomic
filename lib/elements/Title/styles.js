import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';

export default {
    title: {
        boxSizing:'border-box',
        fontFamily:fonts.title,
        color:colors.title,
        fontWeight:'600',
        marginTop:'20px',
        marginBottom:'15px',
        lineHeight:'1.1'
    },
    h1: {
        fontSize:'22px'
    },
    h2: {
        fontSize:'20px',
    },
    h3: {
        fontSize:'18px',
    },
    h4: {
        fontSize:'15px'
    }
};