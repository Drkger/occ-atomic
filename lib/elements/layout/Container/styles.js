import grid from '../../../subatomic/grid';

export default {
    conFluid: {
        boxSizing:'border-box',
        paddingRight:'15px',
        paddingLeft:'15px',
        marginRight:'auto',
        marginLeft:'auto',
        '&:before, &:after': {
            display:'table',
            content:'""'
        },
        '&:after': {
            clear:'both'
        }
    },
    con: {
        extend: 'conFluid',
        margin:'0 auto'
    },
    [`@media (min-width: ${grid.sm}px)`]: {
        con: {
            width:'750px'
        }
    },
    [`@media (min-width: ${grid.md}px)`]: {
        con: {
            width:'970px'
        }
    },
    [`@media (min-width: ${grid.lg}px)`]: {
        con: {
            width:'1170px'
        }
    }
};