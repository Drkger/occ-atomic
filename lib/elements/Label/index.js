import injectSheet from 'react-jss';
import Label from './Label';
import styles from './styles';

export default injectSheet(styles)(Label);