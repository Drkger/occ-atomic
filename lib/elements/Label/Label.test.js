import React from 'react';
import { shallow } from 'enzyme';
import Label from './Label';
import styles from './styles';
import LabelJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Label", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Label classes={classes}>Label</Label>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('shows the children content', () => {
        const wrapper = shallow(
            <Label classes={classes}>Label</Label>
        );
        expect(wrapper.text()).toEqual('Label');
    });

    it('calls onClick function', () => {
        const onClick = jest.fn();
        const wrapper = shallow(
            <Label classes={classes} onClick={onClick}>Label</Label>
        );
        wrapper.simulate('click');
        expect(onClick.mock.calls.length).toBe(1);
    });

    it('calls onClose function', () => {
        const onClose = jest.fn();
        const wrapper = shallow(
            <Label classes={classes} onClose={onClose}>Label</Label>
        );
        wrapper.find('span').simulate('click', { stopPropagation() {} });
        expect(onClose.mock.calls.length).toBe(1);
    });

    it('has the right theme', () => {
        const wrapper = shallow(
            <Label classes={classes} theme="default">Label</Label>
        );
        expect(wrapper.prop('className')).toEqual('label default');
    });

});

describe("LabelJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <LabelJSS>Label</LabelJSS>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Label styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});