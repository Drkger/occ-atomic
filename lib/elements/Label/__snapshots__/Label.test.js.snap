// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    classes={
      Object {
        "close": "close",
        "cursor": "cursor",
        "default": "default",
        "label": "label",
      }
    }
    theme="default"
  >
    Label
  </Label>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Label",
        undefined,
      ],
      "className": "label default",
      "id": undefined,
      "onClick": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      "Label",
      undefined,
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Label",
          undefined,
        ],
        "className": "label default",
        "id": undefined,
        "onClick": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        "Label",
        undefined,
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabelJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Label)
    theme="default"
  >
    Label
  </Jss(Label)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Label",
      "classes": Object {
        "close": "Label-close-0-1-4",
        "cursor": "Label-cursor-0-1-2",
        "default": "Label-default-0-1-3",
        "label": "Label-label-0-1-1",
      },
      "theme": "default",
    },
    "ref": null,
    "rendered": "Label",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Label",
        "classes": Object {
          "close": "Label-close-0-1-4",
          "cursor": "Label-cursor-0-1-2",
          "default": "Label-default-0-1-3",
          "label": "Label-label-0-1-1",
        },
        "theme": "default",
      },
      "ref": null,
      "rendered": "Label",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
